view/ViewableData.php

```diff
@@ -127,6 +134,41 @@ public function __set($property, $value) {
 			$this->setField($property, $value);
 		}
 	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function __call($method, $arguments) {
+		$this->checkAndRedefineMethods();
+		return parent::__call($method, $arguments);
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function hasMethod($method) {
+		$this->checkAndRedefineMethods();
+		return parent::hasMethod($method);
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function allMethodNames($custom = false) {
+		$this->checkAndRedefineMethods();
+		return parent::allMethodNames($custom);
+	}
+
+	/**
+	 * Check whether the underlying failover object has changed or been removed, and call
+	 * defineMethods() if it has
+	 */
+	protected function checkAndRedefineMethods() {
+		if ($this->failover !== $this->oldFailover) {
+			$this->oldFailover = $this->failover;
```

Comment:
I did consider using `spl_object_hash()` for comparing the failover objects (instead of keeping two copies of them in memory), but I don’t think it’s reliable enough - the same hash can be re-used once an object is destroyed.